extensions/amp-selector/0.1/amp-selector.js

```diff
@@ -324,38 +324,32 @@ export class AmpSelector extends AMP.BaseElement {
     if (el.hasAttribute('disabled')) {
       return;
     }
-
     this.mutateElement(() => {
-      /** @type {?Array<string>} */
-      let selectedValues;
       if (el.hasAttribute('selected')) {
         if (this.isMultiple_) {
           this.clearSelection_(el);
-          selectedValues = this.setInputs_();
+          this.setInputs_();
         }
       } else {
         this.setSelection_(el);
-        selectedValues = this.setInputs_();
-      }
-
-      // Don't trigger action or update focus if
-      // selected values haven't changed.
-      if (selectedValues) {
-        // Newly picked option should always have focus.
-        this.updateFocus_(el);
-
-        // Trigger 'select' event with two data params:
-        // 'targetOption' - option value of the selected or deselected element.
-        // 'selectedOptions' - array of option values of selected elements.
-        const name = 'select';
-        const selectEvent =
-            createCustomEvent(this.win, `amp-selector.${name}`, dict({
-              'targetOption': el.getAttribute('option'),
-              'selectedOptions': selectedValues,
-            }));
-        this.action_.trigger(this.element, name, selectEvent,
-            ActionTrust.HIGH);
+        this.setInputs_();
       }
+      // Newly picked option should always have focus.
+      this.updateFocus_(el);
+
+      // Trigger 'select' event with two data params:
+      // 'targetOption' - option value of the selected or deselected element.
+      // 'selectedOptions' - array of option values of selected elements.
+      const name = 'select';
+      const selections =
```

Comment:
Just for my sake of understanding: I assume this line is the real fix because `selectedValues` from previous code only had 1 selected item (the last one that was selected) but now we have all the selected items. So previously `targetOption` and `selectedOptions[0]` were the same thing and `selectedOptions` had the length of `1`. Is that correct?

Reply:
So actually (TIL) that amp-selector creates hidden input fields based on the selected options, but only if the `name` attribute is provided on the `<amp-selector>` itself. That's what `setInputs_()` does. 

So `selectedValues = setInputs_()` returns empty array if `name` attribute is missing, which makes it not very useful. Also seems like `event.selectedOptions` was never publicly documented anywhere.